test/test_suite.sh

```diff
@@ -1,253 +1,276 @@
+  [[ ${lines[3]} == "" ]]
 }
 
-@test "Memory" {
-  run bash -c 'echo ">memory" | nc -v 127.0.0.1 4711'
-  echo "output: ${lines[@]}"
-  [[ ${lines[0]} == "Connection to 127.0.0.1 4711 port [tcp/*] succeeded!" ]]
-  [[ ${lines[1]} =~ "memory allocated for internal data structure:" ]]
-  [[ ${lines[2]} =~ "dynamically allocated allocated memory used for strings:" ]]
-  [[ ${lines[3]} =~ "Sum:" ]]
-  [[ ${lines[4]} == "---EOM---" ]]
-}
-
-@test "Get client ID" {
-  run bash -c 'echo ">clientID" | nc -v 127.0.0.1 4711'
-  echo "output: ${lines[@]}"
-  [[ ${lines[0]} == "Connection to 127.0.0.1 4711 port [tcp/*] succeeded!" ]]
-  [[ ${lines[2]} == "---EOM---" ]]
+@test "Recent blocked" {
+  run bash -c 'echo ">recentBlocked >quit" | nc -v 127.0.0.1 4711'
+  printf "%s\n" "${lines[@]}"
+  [[ ${lines[1]} == "regex5.com" ]]
+  [[ ${lines[2]} == "" ]]
 }
 
-@test "Recent blocked" {
-  run bash -c 'echo ">recentBlocked" | nc -v 127.0.0.1 4711'
-  echo "output: ${lines[@]}"
-  [[ ${lines[0]} == "Connection to 127.0.0.1 4711 port [tcp/*] succeeded!" ]]
-  [[ ${lines[1]} == "addomain.com" ]]
-  [[ ${lines[2]} == "---EOM---" ]]
-}
-
-# @test "IPv6 socket connection" {
-#   run bash -c 'echo ">recentBlocked" | nc -v ::1 4711'
-#   echo "output: ${lines[@]}"
-#   [[ ${lines[0]} == "Connection to ::1 4711 port [tcp/*] succeeded!" ]]
-#   [[ ${lines[1]} == "addomain.com" ]]
-#   [[ ${lines[2]} == "---EOM---" ]]
-# }
-
-@test "DB test: Tables created and populated?" {
-  run bash -c 'sqlite3 pihole-FTL.db .dump'
-  echo "output: ${lines[@]}"
+@test "pihole-FTL.db schema as expected" {
+  run bash -c 'sqlite3 /etc/pihole/pihole-FTL.db .dump'
+  printf "%s\n" "${lines[@]}"
   [[ "${lines[@]}" == *"CREATE TABLE queries ( id INTEGER PRIMARY KEY AUTOINCREMENT, timestamp INTEGER NOT NULL, type INTEGER NOT NULL, status INTEGER NOT NULL, domain TEXT NOT NULL, client TEXT NOT NULL, forward TEXT );"* ]]
   [[ "${lines[@]}" == *"CREATE TABLE ftl ( id INTEGER PRIMARY KEY NOT NULL, value BLOB NOT NULL );"* ]]
   [[ "${lines[@]}" == *"CREATE TABLE counters ( id INTEGER PRIMARY KEY NOT NULL, value INTEGER NOT NULL );"* ]]
-  [[ "${lines[@]}" == *"INSERT INTO \"counters\" VALUES(0,0);"* ]]
-  [[ "${lines[@]}" == *"INSERT INTO \"counters\" VALUES(1,0);"* ]]
-  [[ "${lines[@]}" == *"INSERT INTO \"ftl\" VALUES(0,2);"* ]]
+  [[ "${lines[@]}" == *"CREATE TABLE network ( id INTEGER PRIMARY KEY NOT NULL, ip TEXT NOT NULL, hwaddr TEXT NOT NULL, interface TEXT NOT NULL, name TEXT, firstSeen INTEGER NOT NULL, lastQuery INTEGER NOT NULL, numQueries INTEGER NOT NULL,macVendor TEXT);"* ]]
   [[ "${lines[@]}" == *"CREATE INDEX idx_queries_timestamps ON queries (timestamp);"* ]]
+  [[ "${lines[@]}" == *"CREATE UNIQUE INDEX network_hwaddr_idx ON network(hwaddr);"* ]]
 }
 
-@test "Arguments check: Invalid option" {
-  run bash -c './pihole-FTL abc'
-  echo "output: ${lines[@]}"
+@test "Fail on invalid argument" {
+  run bash -c './pihole-FTL-linux-x86_64 abc'
```

Comment:
Can we use some variable instead of hardcoding the name? This makes local testing harder, as the binary is not named this way by default.